### ISSUE TITLE
Segfault with define_constant(float("nan"))

### ISSUE THREAD

**Issue by IvanYashchuk** [NVIDIA/Fuser#122]
There's a crash with `nan` constants for some reason, `float("inf")` works!
```py
from nvfuser import FusionDefinition, DataType
import torch

def nvfuser_fusion_id(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(symbolic_sizes=[-1], contiguous=[True], dtype=DataType.Double, is_cpu=False)
    c0 = fd.define_constant(float("nan"))
    T1 = fd.ops.add(T0, c0)
    fd.add_output(T1)

with FusionDefinition() as fd:
    nvfuser_fusion_id(fd)

a = torch.randn(8, dtype=torch.double, device='cuda')
fd.execute([a])
```

**Comment by kevinstephano** [comment]
Hmm, looks like we might have walked off a TrieNode or something.

```
0x00007fff6339df3d in nvfuser::python_frontend::FusionCache::createChild(nvfuser::python_frontend::TrieNode*, nvfuser::python_frontend::RecordFunctor*) () from /opt/pytorch/pytorch/torch/lib/libnvfuser_codegen.so
```

**Comment by kevinstephano** [comment]
The problem is that the `Constant` Fusion Record's equality operator has to be special cased to match a value of `nan`.  When the Record is looked up with a value of `nan` it returns `False` for a match even though the hash map entry was created.

**Comment by kevinstephano** [comment]
The reason for the segfault is the node returned after lookup from the hash map returns a `nullptr` that I did not check for in the right spot due to the fact that `nan` equality returns `False`.  The check should have been put one line up.

https://github.com/NVIDIA/Fuser/blob/7b24fa8fc048b6f4138067d542f274584721b436/csrc/python_frontend/fusion_cache.cpp#L255-L258

**Comment by kevinstephano** [comment]
Fixed with PR #144.
